### PR TITLE
include column lineage in dataset resource

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
Signed-off-by: Pawel Leszczynski <leszczynski.pawel@gmail.com>

### Problem

Column lineage should be returned within `dataset` resource.

Closes: #2113

### Solution

 * create method in `ColumnLineageService` which enriches `Dataset` with column lineage information, 
 * use the method in `DatasetResource`.

> **Note:** All database schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've updated the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: api/src/main/java/marquez/db/ColumnLineageDao.java [hunk 1]
```diff
@@ -150,4 +150,47 @@ Set<ColumnLineageNodeData> getLineage(
...
+          output_fields.field_name,
+          output_fields.type,
+          c.transformation_description,
+          c.transformation_type,
+          c.created_at,
+          c.updated_at
+      """)
+  Set<ColumnLineageNodeData> getLineageRowsForDatasets(
+      @BindBeanList(
+              propertyNames = {"left", "right"},
```
Comment (wslulciuc) [inline api/src/main/java/marquez/db/ColumnLineageDao.java:193]: Very cool! But, how is the `left` and `right` properties used in this context?

Comment (pawel-big-lebowski) [reply api/src/main/java/marquez/db/ColumnLineageDao.java:193]: these are the properties of `Pair`. `propertyNames` tell which values should be put to SQL.

File: api/src/main/java/marquez/service/models/ColumnLineageInputField.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+@EqualsAndHashCode
+@ToString
+@Getter
+@AllArgsConstructor
+public class ColumnLineageInputField {
+  @NotNull private String datasetNamespace;
```
Comment (pawel-big-lebowski) [inline api/src/main/java/marquez/service/models/ColumnLineageInputField.java:19]: TODO: Rename it to `namespace`, `dataset` and `field`

File: api/src/test/java/marquez/DatasetIntegrationTest.java [hunk 3]
```diff
@@ -440,4 +443,49 @@ public void testApp_showsDeletedDatasetAfterReceivingNewVersion() throws IOExcep
...
+    assertThat(columnLineage.get(0).getInputFields()).hasSize(2);
+
+    // verify getDataset returns non-empty column lineage
+    columnLineage = client.getDataset("namespace", "dataset_b").getColumnLineage();
+    assertThat(columnLineage).hasSize(1);
+    assertThat(columnLineage.get(0).getInputFields()).hasSize(2);
+
+    client.deleteJob("namespace", "job_name");
+    client.deleteDataset("namespace", "dataset_a");
+    client.deleteDataset("namespace", "dataset_b");
```
Comment (mobuchowski) [inline api/src/test/java/marquez/DatasetIntegrationTest.java:489]: I think we should have `@BeforeEach` method cleaning tables as we have in different integration tests.

**Review by mobuchowski — APPROVED** [review]
Few minor nits, besides LGTM

**Pull request merged**